libsolidity/codegen/ABIFunctions.h

```diff
@@ -253,6 +253,16 @@ class ABIFunctions
 	/// @returns the size of the static part of the encoding of the given types.
 	static size_t headSize(TypePointers const& _targetTypes);
 
+	/// @returns a string containing a comma-separated list of variable names consisting of @a _baseName suffixed
+	/// with increasing integers in the range [@a _startSuffix, @a _endSuffix), if @a _startSuffix < @a _endSuffix,
+	/// and with decreasing integers in the range [@a _endSuffix, @a _startSuffix), if @a _endSuffix < @a _startSuffix.
+	/// If @a _startSuffix == @a _endSuffix, the empty string is returned.
+	static std::string suffixedVariableNameList(std::string const& _baseName, size_t _startSuffix, size_t _endSuffix);
+
+	/// @returns the number of variables needed to store a type.
+	/// This is one for any type other than a dynamically sized calldata array, for which it is two.
```

Comment:
This is only true if options.encodeFunctionFromStack is false. If it is true, then function types also need two variables.